numpy/core/src/multiarray/einsum.c.src

```diff
@@ -2078,17 +2078,6 @@ get_single_op_view(PyArrayObject *op, int  iop, char *labels,
         if (*ret == NULL) {
             return -1;
         }
-        if (!PyArray_Check(*ret)) {
-            Py_DECREF(*ret);
-            *ret = NULL;
-            PyErr_SetString(PyExc_RuntimeError,
-                        "NewFromDescr failed to return an array");
```

Comment:
If `PyArray_New` doesn't return an array, I think it's safe to say all bets are off

Reply:
It would be an interesting exercise to try and define a new `PyArrayObject` that does not have `PyArray_Type` as a base class but respects the NumPy `PyArrayObject_fields` layout and can traverse `PyArray_NewFromDescr_int` safely. I agree it is not realistic.